XenModel/PoolJoinRules.cs

```diff
@@ -130,6 +131,9 @@ public static Reason CanJoinPool(IXenConnection slaveConnection, IXenConnection
 
             if (LicenseMismatch(slaveHost, masterHost))
                 return Reason.LicenseMismatch;
+
+            if (UnlicensedMasterMaxNumberHostReached(masterConnection, masterHost))

```

Comment:
If this function take masterConnection as parameter, then masterHost shouldn't be needed